src/TrimAttributes.LinkDescription.xml

```diff
@@ -0,0 +1,291 @@
+<linker>
+  <!--
+    This file contains attributes we've historically removed.
+
+    ILLink supported an "aggressive attribute trimming" mode, which would
+    remove the attributes listed herein, but that was removed (https://github.com/xamarin/xamarin-macios/issues/20906)
+
+    However, this results in a significant size hit on mobile apps, so we've
+    re-implemented the aggressive attribute removal, under a different feature
+    switch (the 'MobileAggressiveAttributeTrimming' MSBuild property).
+
+    The list of attributes to remove comes from the following files:
+
+    * https://github.com/dotnet/runtime/blob/d22c74dc974a757b48380a01dd0450efba31db64/src/mono/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.xml#L4-L16
+    * https://github.com/dotnet/runtime/blob/d22c74dc974a757b48380a01dd0450efba31db64/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.Shared.xml#L93-L350
+  -->
+
+  <!-- The attributes listed here are only removed on mono. Removing them on coreclr would change runtime behavior. -->
+  <!-- https://github.com/dotnet/runtime/blob/d22c74dc974a757b48380a01dd0450efba31db64/src/mono/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.xml#L4-L16 -->
+  <assembly fullname="System.Private.CoreLib" feature="ObjCRuntime.AggressiveAttributeTrimmingMono" featurevalue="true">
+    <type fullname="System.Runtime.CompilerServices.IntrinsicAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+  </assembly>
+  <assembly fullname="*" feature="ObjCRuntime.AggressiveAttributeTrimmingMono" featurevalue="true">
+    <!-- Internal attributes shared as implementation, so they could be in any assembly -->
+    <type fullname="System.Runtime.Versioning.NonVersionableAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+  </assembly>
+
+  <!-- https://github.com/dotnet/runtime/blob/d22c74dc974a757b48380a01dd0450efba31db64/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.Shared.xml#L93-L350 -->
+  <assembly fullname="System.Private.CoreLib" feature="ObjCRuntime.AggressiveAttributeTrimming" featurevalue="true">
+    <!-- System -->
+    <type fullname="System.CLSCompliantAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.ObsoleteAttribute">
+      <!--
+        Note that removing this attribute can change runtime behavior. For example,
+        System.Xml.Serialization will behave differently if a ctor is Obsolete.
+        This is low enough risk on when 'ObjCRuntime.AggressiveAttributeTrimming' is enabled to justify
+        removing the attribute for size savings. The app developer can override this setting
+        to keep all ObsoleteAttributes.
```

Comment:
Should we document this somewhere outside this file? /docs maybe? and maybe link it from [build-properties.md](https://github.com/xamarin/xamarin-macios/pull/21314/files#diff-1b3063186e01c6aea9b98fdfb003e88e70a8858520bb54060cf9834d8751c05f) entry you added